src/CommandLineUtils/CommandLineApplication.cs

```diff
@@ -29,7 +29,7 @@ public partial class CommandLineApplication : IServiceProvider, IDisposable
         private List<Action<ParseResult>> _onParsingComplete;
         internal readonly Dictionary<string, PropertyInfo> _shortOptions = new Dictionary<string, PropertyInfo>();
         internal readonly Dictionary<string, PropertyInfo> _longOptions = new Dictionary<string, PropertyInfo>();
-        private readonly HashSet<string> _names = new HashSet<string>(StringComparer.Ordinal);
```

Comment:
Rather than add a new API, I think I would rather just keep the code as is, but change this to `StringComparer.OrdinalIgnoreCase`. This was a new feature in 2.3 which hasn't seen much usage yet. Now that I think about it, I doubt many users would want subcommands to vary by case, and in 2.2 subcommand name matching was already case-insensitive.